### PR TITLE
Switch to Dune / jbuilder

### LINKED ISSUE #12: Support dylib
See https://github.com/paurkedal/ocaml-caqti/issues/12

They  said that hte META file is missing plugin(byte) and plugin(native) stanzas, but I think we also need to add a build for a .cmxs file, which I don't really know how to do.

One recommended solution is to use Dune/jbuilder instead of whatever we currently use to make the META file: https://github.com/paurkedal/ocaml-caqti/issues/12#issuecomment-379544209

### PR DESCRIPTION
**Pull request by brendanlong**
Add Dune's jbuild files and remove everything we don't need anymore.

I added examples/ct_example to the build but wasn't able to make
examples/dbi_example work.

I also removed files related to Godi since it seems to be discontinued
since 2014 so converting them didn't seem worthwhile.

Note: This build will probably work on earlier versions of jbuilder
but I set the dependency to the one I've actually tested.

Fixes #12

### CODE REVIEW

**Comment by brendanlong** [comment]
Confirmed that pinning this makes it loadable as a dynlib:

https://circleci.com/gh/brendanlong/ocaml-caqti/38

(This CI build is still failing for other reasons, but the dylib loads now)

File: src/jbuild [hunk 1]
```diff
@@ -0,0 +1,8 @@
+(jbuild_version 1)
+
+(library
+ ((name freetds)
+  (public_name freetds)
+  (wrapped true)
```
Comment (Chris00) [inline src/jbuild]: This is the default, thus not needed.

**Comment by Chris00** [comment]
It seems the detection of the library is not done anymore so they necessarily must be at a standard location.

**Comment by brendanlong** [comment]
Yeah.. it seems like the sqlite3 library does something fairly complicated to detect the library location. I'll see if I can do something like that here.

**Comment by brendanlong** [comment]
There's instructions for using pkg-config here: http://jbuilder.readthedocs.io/en/latest/quick-start.html#defining-a-library-with-c-stubs-using-pkg-config

But I don't think pkg-config supports FreeTDS?

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,16 @@
+all: build
+
+build:
+	@jbuilder build @install
+
+clean:
+	@jbuilder clean
+	@rm -f **/.merlin
```
Comment (Chris00) [inline Makefile]: $(RM)

Comment (brendanlong) [reply Makefile]: Fixed

Comment (brendanlong) [reply Makefile]: Fixed

**Comment by Chris00** [comment]
> pkg-config supports FreeTDS

It seems indeed that freetds has no pkg-config support.  MacOS CI shpuld be added to make sure it works on that platform as well.  Can you take that point?

**Comment by brendanlong** [comment]
My work computer is a Mac and the jbuilder build works. I'll see if I can figure out how to make TravisCI build it too.

**Comment by Chris00** [comment]
Thanks.

**Pull request merged**